diff --git a/src/bitcoinrpc/connection.py b/src/bitcoinrpc/connection.py
index 960cf5d..79e04bf 100644
--- a/src/bitcoinrpc/connection.py
+++ b/src/bitcoinrpc/connection.py
@@ -67,7 +67,35 @@ class BitcoinConnection(object):
             return self.proxy.getblockcount()
         except JSONRPCException,e:
             raise _wrap_exception(e.error)    
-        
+
+    def getmemorypool(self):
+        """sign"""
+        try:
+            return self.proxy.getmemorypool()
+        except JSONRPCException,e:
+            raise _wrap_exception(e.error)  
+
+    def importtransaction(self, tx):
+        """sign"""
+        try:
+            return self.proxy.importtransaction(tx)
+        except JSONRPCException,e:
+            raise _wrap_exception(e.error)  
+
+    def signmessage(self, address,message):
+        """sign"""
+        try:
+            return self.proxy.signmessage(address, message)
+        except JSONRPCException,e:
+            raise _wrap_exception(e.error)  
+
+    def verifymessage(self, address, sig, message):
+        """verif"""
+        try:
+            return self.proxy.verifymessage(address, sig, message )
+        except JSONRPCException,e:
+            raise _wrap_exception(e.error)  
+
     def getblocknumber(self):
         """
         Returns the block number of the latest block in the longest block chain.
