diff -u -r bitcoin/src/main.cpp bitcoin-electrum/src/main.cpp
--- bitcoin/src/main.cpp	2011-12-28 21:23:40.844812872 +0200
+++ bitcoin-electrum/src/main.cpp	2011-12-28 17:47:27.000000000 +0200
@@ -2965,13 +2965,16 @@
 
             // Size limits
             unsigned int nTxSize = ::GetSerializeSize(tx, SER_NETWORK);
-            if (nBlockSize + nTxSize >= MAX_BLOCK_SIZE_GEN)
-                continue;
+            //if (nBlockSize + nTxSize >= MAX_BLOCK_SIZE_GEN)
+            //    continue;
 
             // Transaction fee required depends on block size
             bool fAllowFree = (nBlockSize + nTxSize < 4000 || CTransaction::AllowFree(dPriority));
             int64 nMinFee = tx.GetMinFee(nBlockSize, fAllowFree, GMF_BLOCK);
 
+	    // Electrum server: do not check fees
+	    nMinFee = 0;
+
             // Connecting shouldn't fail due to dependency on other memory pool transactions
             // because we're already processing them in order of dependency
             map<uint256, CTxIndex> mapTestPoolTmp(mapTestPool);
@@ -2981,8 +2984,8 @@
             int nTxSigOps = 0;
             if (!tx.ConnectInputs(mapInputs, mapTestPoolTmp, CDiskTxPos(1,1,1), pindexPrev, nFees, false, true, nTxSigOps, nMinFee))
                 continue;
-            if (nBlockSigOps + nTxSigOps >= MAX_BLOCK_SIGOPS)
-                continue;
+            //if (nBlockSigOps + nTxSigOps >= MAX_BLOCK_SIGOPS)
+            //    continue;
             swap(mapTestPool, mapTestPoolTmp);
 
             // Added
